core/server/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -199,8 +198,9 @@ public void start() throws IOException {
     getExecutorService().submit(mSessionCleaner);
 
     // Start the space reserver
-    if (mSpaceReserver != null) {
-      getExecutorService().submit(mSpaceReserver);
+    if (spaceReserver != null) {
```

Comment:
Is `spaceReserver` allowed to be created here? Then, we could just check for the configuration here, create the `SpaceReserver`, and directly add it to the threadpool.